app/src/main/java/org/mifos/mobile/viewModels/RecentTransactionViewModel.kt

```diff
@@ -28,24 +28,34 @@ class RecentTransactionViewModel @Inject constructor(private val recentTransacti
         loadRecentTransactions(offset, limit)
     }
 
-    private fun loadRecentTransactions(offset: Int, limit: Int) {
+    private fun loadRecentTransactions(offset: Int, limit: Int, refresh: Boolean = false) {
         viewModelScope.launch {
-            _recentTransactionUiState.value = RecentTransactionUiState.Loading
-            recentTransactionRepositoryImp.recentTransactions(offset, limit).catch {
-                _recentTransactionUiState.value =
-                    RecentTransactionUiState.Error(R.string.recent_transactions)
-            }.collect {
-                if (it.totalFilteredRecords == 0) {
-                    _recentTransactionUiState.value = RecentTransactionUiState.EmptyTransaction
-                } else if (loadmore && it.pageItems.isNotEmpty()) {
-                    _recentTransactionUiState.value =
-                        RecentTransactionUiState.LoadMoreRecentTransactions(it.pageItems)
-                } else if (it.pageItems.isNotEmpty()) {
+            try {
+                _recentTransactionUiState.value = RecentTransactionUiState.Loading
+
+                recentTransactionRepositoryImp.recentTransactions(offset, limit).catch {
+
                     _recentTransactionUiState.value =
-                        RecentTransactionUiState.RecentTransactions(it.pageItems)
+                        RecentTransactionUiState.Error(R.string.recent_transactions)
+                }.collect {
+                    if (it.totalFilteredRecords == 0) {
+
+                        _recentTransactionUiState.value = RecentTransactionUiState.EmptyTransaction
+                    } else if (loadmore && it.pageItems.isNotEmpty()) {
+
+                        _recentTransactionUiState.value =
+                            RecentTransactionUiState.LoadMoreRecentTransactions(it.pageItems)
+                    } else if (it.pageItems.isNotEmpty()) {
+
+                        _recentTransactionUiState.value =
+                            RecentTransactionUiState.RecentTransactions(it.pageItems)
+                    }
                 }
+            } finally {
+                _recentTransactionUiState.value = RecentTransactionUiState.Initial
```

Comment:
any reason to use finally here ? This would mean that you would always want to execute this statement

Reply:
are you certain that the error arises from the viewmodel and not from the fragment ?

Reply:
Yes!  here are the log statements before finally..!
![Screenshot 2024-01-05 180450](https://github.com/openMF/mifos-mobile/assets/136256522/a79a927f-c983-4e7e-8703-39f637e825e1)

Reply:
And after finally @PratyushSingh07 
![Screenshot 2024-01-05 180214](https://github.com/openMF/mifos-mobile/assets/136256522/a184569c-86d2-4a83-9007-09f5538c097d)